controlpanel/api/cluster.py

```diff
@@ -218,230 +110,68 @@ def url(self):
         return f"https://{ingresses[0].spec.rules[0].host}"
 
 
-class IAMRole:
-
-    def __init__(self, role_name):
-        self.policy = {
-            "Version": "2012-10-17",
-            "Statement": [
-                {
-                    "Effect": "Allow",
-                    "Principal": {
-                        "Service": "ec2.amazonaws.com",
-                    },
-                    "Action": "sts:AssumeRole",
-                },
-                {
-                    "Effect": "Allow",
-                    "Principal": {
-                        "AWS": iam_arn(f"role/{settings.K8S_WORKER_ROLE_NAME}"),
-                    },
-                    "Action": "sts:AssumeRole",
-                },
-            ],
-        }
-        self.role_name = role_name
-
-    def save(self):
-        try:
-            aws.create_role(self.role_name, self.policy)
-        except Exception as e:
-            if not is_ignored_exception(e):
-                raise e
-
-    def allow_assume_role_with_saml(self):
-        self.policy["Statement"].append(dict(SAML_STATEMENT))
-
-    def allow_assume_role_with_oidc(self, sub):
-        stmt = dict(OIDC_STATEMENT)
-        stmt["Condition"] = {
-            "StringEquals": {
-                f"{settings.OIDC_DOMAIN}/:sub": sub,
-            }
-        }
-        self.policy["Statement"].append(stmt)
-
-
-class S3AccessPolicy(dict):
-    POLICY_NAME = "s3-access"
+class Bucket:
+    """Wraps a S3Bucket model to provide convenience methods for AWS"""
+    def __init__(self, bucket):
+        self.bucket = bucket
 
-    def __init__(self, role_name=None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        self.role_name = role_name
-        self._ro = set()
-        self._rw = set()
-
-        if "Version" not in self:
-            self["Version"] = "2012-10-17",
-
-        for statement in self.get("Statement", []):
-            sid = statement.get("Sid")
-
-            if sid == "readonly":
-                self._ro = set([
-                    arn.rstrip('/*')
-                    for arn in statement["Resource"]
-                ])
+    @property
+    def arn(self):
+        return s3_arn(self.bucket.name)
 
-            elif sid == "readwrite":
-                self._rw = set([
-                    arn.rstrip('/*')
-                    for arn in statement["Resource"]
-                ])
+    def create(self):
+        return aws.create_bucket(self.bucket.name, self.bucket.is_data_warehouse)
 
-    @classmethod
-    def load(cls, role_name):
-        document = aws.get_inline_policy_document(
-            role_name=role_name,
-            policy_name=cls.POLICY_NAME,
-        )
 
-        if document is None:
-            return cls(role_name=role_name)
-
-        policy = cls(role_name=role_name, **document)
-        return policy
+class Group:
+    """
+    Represents a group of cluster users who have access to datasources
```

Comment:
As mentioned in [other comment](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/764#discussion_r336036520) we should probably elaborate on what it is and why we did it, e.g.:

> A Group uses a managed policy as a way to group IAM roles that have access to same resources.
>
> This is done to overcome IAM limitation which doesn't allow adding IAM roles to IAM groups: https://stackoverflow.com/a/48087433/455642

or something like that, maybe even mentioned that each Analytical Platform user has an IAM role which is assumed by their tools - but that's probably overkill and potentially "basic knowledge".